artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/CheckDependencies.java

```diff
@@ -54,4 +54,17 @@ public static final boolean isKQueueAvailable() {
          return false;
       }
    }
+
+   public static boolean isIoUringAvailable() {
+      try {
+         return Env.isLinuxOs() && (boolean) (Class.forName("io.netty.incubator.channel.uring.IOUring")
```

Comment:
Should just call IOUring.isAvailable() like we do with kqueue and epoll, dont do this via reflection

Reply:
It's not available via maven afaik so it's up to the users to compile it from source and use it

Reply:
<dependency> <groupId>io.netty.incubator</groupId> <artifactId>netty-incubator-transport-native-io_uring</artifactId> <version>0.0.3.Final</version> <classifier>linux-x86_64</classifier> </dependency>

Reply:
Its available. These is actually in the docs on the netty github

Reply:
In fact 0.0.4 is latest

https://mvnrepository.com/artifact/io.netty.incubator/netty-incubator-transport-native-io_uring/0.0.4.Final

Reply:
Wow ok, will remove the reflection then, thanks to have checked!